hamp_radar/iquick.py

```diff
@@ -233,6 +232,159 @@ def decode_time(ds):
+            ds1.to_zarr(fout)
+
+            ds2 = (radar['FFTD'].to_dataset(dim='cocx')
+                   .assign_coords(pulse_time=lambda ds2: time_frame + dtfft)
+                   .stack(pulse_time=["frame", "fft"], create_index=False)
+                   .transpose("iq","range","pulse_time")
+                   .rename({0:"co"}).rename({1:"cx"})
+                   .assign_coords(range=("range",np.arange(ngate)*len_gate+d_gate1))
+                   .assign_coords(iq=("iq",["i","q"]))
+              )
+            ds2["range"].attrs["long_name"] = "distance from aircraft"
+            ds2["range"].attrs["units"] = "m"
+            ds2["iq"].attrs["long_name"] = "signal phase (inphase or quadrature)"
+            fout  = f'{zarr_path}/HALO-{pd.to_datetime(radar.time[0].values):%Y%m%da-%H%M%S}-cocx.zarr'
+            ds2.to_zarr(fout)
+        else:
+            sys.exit('error frames do not increment monotonically')
+    return ds_cocx,ds_frms
+
+
+def merge_iqf(cocx_files,frms_files):
+
+    """
+    Takes a list of pulse data (cocx_files) and rame data (frame_files)
+    merges them into a single data set including additional metadata, which
+    the function returns, ideally to be written to a zarr dataset.
+
+    cocx_files: list of zarr files with pulse data
+    frms_files: list of zarr files with frame data
+
+    ex.,
+    zarr_path = './radar_zarr_files'
+    files = [pds_path + x for x in sorted(os.listdir(path))]
+    untangle_iqf(files,zarr_path)
+
+    cocx_files = sorted(glob.glob(zarr_path + '*cocx.zarr'))
+    frms_files = sorted(glob.glob(zarr_path + '*frms.zarr'))
+    ds = merge_radar_zarr(cocx_files,frms_files)
+    ds.to_zarr(fout)
+    """
+    import numpy as np
+    import xarray as xr
+    ds_frms = xr.open_mfdataset(frms_files,engine='zarr')
+    dx = (ds_frms
+          .merge(ds_frms.NPw.to_dataset(dim='cocx').rename({0:"co_NPw"}).rename({1:"cx_NPw"}))
+          .merge(ds_frms.CPw.to_dataset(dim='cocx').rename({0:"co_CPw"}).rename({1:"cx_CPw"}))
+          .drop_vars(["NPw","CPw"])
+    )
+    dx["co_NPw"].attrs['long_name'] = 'co-channel noise power pin-mod in save position'
+    dx["cx_NPw"].attrs['long_name'] = 'cross channel noise power pin-mod in save position'
+    dx["co_CPw"].attrs['long_name'] = 'co-channel noise power int. source'
+    dx["cx_CPw"].attrs['long_name'] = 'cross channel noise power int. source'
+    dx["frame_time"].attrs['long_name'] = "radar timestamp for frame"
+
+    dy= xr.open_mfdataset(cocx_files,engine='zarr')
+    dy["co"].attrs['long_name'] = 'iq data from co-polarazied receiver'
+    dy["cx"].attrs['long_name'] = 'iq data from cross-polarazied receiver'
+    dy["pulse_time"].attrs['long_name'] = 'estimated time of pulse from calculated prf'
+    dy["range"].attrs["long_name"] = "distance from aircraft"
+    dy["range"].attrs["units"] = "m"
+    dy["iq"].attrs["long_name"] = "signal phase (inphase or quadrature)"
```

Comment:
Those attributes should also be assigned when creating the variables (just as in the comment for the function above).